src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,2464 @@
+            auto const seq = env.seq(alice);
+            env(batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                batch::inner(ammCreate(XRP(10), USD(10)), seq + 1),
+                batch::inner(pay(alice, bob, XRP(10)), seq + 2),
+                ter(telINSUF_FEE_P));
+            env.close();
+        }
+    }
+
+    void
+    testBadInnerFee(FeatureBitset features)
+    {
+        testcase("bad inner fee");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+        XRPAmount const feeDrops = env.current()->fees().base;
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const carol = Account("carol");
+        auto const gw = Account("gw");
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(1000), alice, bob, carol, gw);
+        env.close();
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        auto const preAlice = env.balance(alice);
+        auto const preBob = env.balance(bob);
+
+        auto const seq = env.seq(alice);
+        auto const batchFee = calcBatchFee(env, 0, 2);
+        auto tx1 = pay(alice, bob, XRP(1));
+        tx1[jss::Fee] = to_string(feeDrops);
+        tx1[jss::Sequence] = seq + 1;
+        tx1[jss::SigningPubKey] = "";
+        tx1[jss::Flags] = tx1[jss::Flags].asUInt() | tfInnerBatchTxn;
+        env(batch::outer(alice, seq, batchFee, tfAllOrNothing),
+            batch::inner_nofill(tx1),
+            batch::inner(pay(alice, bob, XRP(2)), seq + 2),
+            ter(tesSUCCESS));
+        auto const txIDs = env.tx()->getBatchTransactionIDs();
+        TxID const parentBatchId = env.tx()->getTransactionID();
+        std::vector<TestBatchData> testCases = {};
+        env.close();
+        validateBatch(env, parentBatchId, testCases);
+
+        // Alice pays fee and sequence; Bob should not be affected.
+        BEAST_EXPECT(env.seq(alice) == seq + 1);
+        BEAST_EXPECT(env.balance(alice) == preAlice - batchFee);
+        BEAST_EXPECT(env.balance(bob) == preBob);
+    }
+
+    void
+    testChangesBetweenViews(FeatureBitset features)
+    {
+        testcase("changes between views");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+        auto const reserve = env.current()->fees().reserve;
+        auto const incReserve = env.current()->fees().increment;
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const carol = Account("carol");
+        env.fund(reserve + incReserve, alice, bob, carol);
+        env.close();
+
+        auto const preAlice = env.balance(alice);
+        auto const preBob = env.balance(bob);
+
+        // Using reserve to create insufficient reserve result
+        auto const batchFee = reserve;
```

Comment:
ditto, should use `calcBatchFee - whatever insufficient amt `

Reply:
In this case we are using the fee to burn them past the reserve so that the transaction fails with a `tecINSUFFICIENT_RESERVE`...

Reply:
I delete this test as its a duplicate of the Batch Type tests which confirm the same behavior